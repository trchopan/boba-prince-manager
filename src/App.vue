<template>
  <div id="app">
    <InitMessage v-if="loading" />
    <Navigation v-else>
      <router-view/>
    </Navigation>
  </div>
</template>

<script>
import { mapGetters } from "vuex";
import InitMessage from "@/components/InitMessage";
import Navigation from "@/components/Navigation";

export default {
  name: "App",
  components: {
    Navigation,
    InitMessage
  },
  computed: mapGetters({ loading: "auth/loading" })
};
</script>

<style lang="scss">
@import url("./assets/style.scss");
</style>
