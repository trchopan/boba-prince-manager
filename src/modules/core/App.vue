<template>
  <div id="app">
    <InitMessage v-if="loading && !profile"/>
    <v-app v-else :dark="darkTheme">
      <Toolbar/>
      <NavigationDrawer/>
      <v-content>
        <v-container fluid :class="{ 'white--text': darkTheme }">
          <transition name="fade" mode="out-in">
            <router-view/>
          </transition>
        </v-container>
      </v-content>
    </v-app>
  </div>
</template>

<script>
import { mapGetters } from "vuex";
import InitMessage from "./components/InitMessage.vue";
import Toolbar from "./components/Toolbar.vue";
import NavigationDrawer from "./components/NavigationDrawer.vue";

export default {
  name: "App",
  components: {
    Toolbar,
    NavigationDrawer,
    InitMessage
  },
  computed: mapGetters({
    darkTheme: "layout/darkTheme",
    loading: "auth/loading",
    profile: "auth/profile"
  }),
  created() {
    this.$store.dispatch("layout/initLocalStorage");
    if (this.$vuetify.breakpoint.lgAndUp) {
      this.$store.dispatch("layout/toggleDrawer");
    }
    this.$t();
  }
};
</script>

<style lang="scss" src="@/style.scss" />
