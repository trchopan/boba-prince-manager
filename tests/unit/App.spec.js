import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import App from "@/App.vue";
import { mockCustomElements } from "@tests/unit/__mocks__/custom-elements.js";
import { docData } from "@tests/unit/__mocks__/firebase-results.js"

const localVue = createLocalVue();
localVue.use(Vuex);

describe("@/App.vue", () => {
  let store;
  let state;
  let getters;
  let actions;
  let wrapper;

  beforeEach(() => {
    state = {
      loading: false,
      profile: null
    };
    getters = {
      "layout/darkTheme": () => true,
      "auth/loading": state => state.loading,
      "auth/profile": state => state.profile
    };
    actions = {
      "layout/initLocalStorage": jest.fn(),
      "layout/toggleDrawer": jest.fn()
    };
    store = new Vuex.Store({
      state,
      getters,
      actions
    });
    wrapper = shallowMount(App, {
      stubs: mockCustomElements,
      mocks: { $vuetify: { breakpoint: { lgAndUp: true } } },
      store,
      localVue
    });
  });
  it("initializes", () => {
    expect(actions["layout/initLocalStorage"]).toBeCalled();
    expect(actions["layout/toggleDrawer"]).toBeCalled();
  });
  it("renders InitMessage correctly", () => {
    let initPage;
    state.loading = true;
    state.profile = null;
    initPage = wrapper.find("initmessage-stub");
    expect(initPage.exists()).toBe(true);
    state.loading = true;
    state.profile = docData;
    initPage = wrapper.find("initmessage-stub");
    expect(initPage.exists()).toBe(false);
  });
  it("matchs snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
