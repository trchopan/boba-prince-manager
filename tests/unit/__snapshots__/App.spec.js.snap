// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/App.vue matchs snapshot 1`] = `
<div
  id="app"
>
  <div
    id="loading"
  >
    <p>
      Loading...
    </p>
  </div>
</div>
`;

exports[`@/App.vue matchs snapshot 2`] = `
<div
  id="app"
>
  <router-view-stub />
</div>
`;
